src/ziggurat/messaging/consumer.clj

```diff
@@ -25,18 +25,23 @@
   [message topic-entity]
   (try
     (s/validate mp/message-payload-schema message)
-    (catch Exception e
+    (catch Exception _
       (log/info "old message format read, converting to message-payload: " message)
       (let [retry-count (or (:retry-count message) 0)
             message-payload (mp/->MessagePayload (dissoc message :retry-count) (keyword topic-entity))]
         (assoc message-payload :retry-count retry-count)))))
 
+(defn- try-deserialize-message
+  [topic-entity-name message]
+  (let [from-proto (mw/deserialize-message message MessagePayloadProto$MessagePayload topic-entity-name)]
+    (if (nil? from-proto) (nippy/thaw message) from-proto)))
+
 (defn convert-and-ack-message
   "De-serializes the message payload (`payload`) using `nippy/thaw` and converts it to `MessagePayload`. Acks the message
   if `ack?` is true."
-  [ch {:keys [delivery-tag] :as meta} ^bytes payload ack? topic-entity]
+  [ch {:keys [delivery-tag]} ^bytes payload ack? topic-entity]
   (try
-    (let [message (nippy/thaw payload)]
+    (let [message (try-deserialize-message topic-entity payload)]
       (when ack?
         (lb/ack ch delivery-tag))
       (convert-to-message-payload message topic-entity))
```

Comment:
Once this method is removed at L19, we can simply return the message at this point.

Reply:
done